pkg/gofr/datasource/error.go

```diff
@@ -0,0 +1,42 @@
+package datasource
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// DBErr represents an error specific to database operations.
+type DBErr struct {
```

Comment:
can be renamed to error such that user can use `datasource.Error`

but it might restrict us with the name, and adding a new error would be with a different naming convention.

Also it can be unexported as user will be using the New Function